lib/utils/compose_ts.ts

```diff
@@ -1314,12 +1316,23 @@ export async function deployProject(
 
 	let $release: Release;
 	try {
+		let contractContent: string | undefined;
+		let contract = await tryGetContractContent(`${projectPath}/balena.yaml`);
```

Comment:
* Not a big deal, but is there a spec or requirement or precedent or anticipated use case or such to support the `.yaml` file extension in addition to `.yml`? If not, I suggest accepting `.yml` only. Otherwise, it's just extra logic to maintain over time. Less code, less chance of bugs.
* It might work either way, but the preferred way of dealing with paths in the CLI codebase -- considering we support Windows as well -- is to use `path.join(projectPath, 'balena.yml')`. On Windows, this will insert a backslash instead of forward slash as path separator.

Reply:
All documentation and code I have found for this accepts `balena.yml` only. So I think it makes sense to maintain that here.

https://www.balena.io/docs/learn/deploy/deploy-with-balena-button/#balenayml-configuration-file
https://github.com/balena-io/balena-ui/blob/998d0ffbe8465447c960dc7472c204a4108a0cac/src/utils/balenaConfig.ts#L206-L234

Reply:
Also https://github.com/balena-io/balena-ui/blob/998d0ffbe8465447c960dc7472c204a4108a0cac/src/utils/balenaConfig.ts#L116-L156

Reply:
Sorry about all the comments, just leaving some context to pick this up later

Reply:
Great comments 👍

Reply:
Just FYI, the reason I went with both extensions was that is how the builder does it.  So the builder could technically take either.  It seems right to keep it consistent?

Reply:
@toochevere I'd opt to remove the `.yaml` extension from the builder since it makes things simpler and consistent all around and sounds like it shouldn't have any dire consequences. I'll ask on FD to see if there is some definite information about `balena.yml`